pyodide-build/pyodide_build/tests/test_build_env.py

```diff
@@ -0,0 +1,106 @@
+import os
+
+import pytest
+
+from conftest import ROOT_PATH
+from pyodide_build import build_env
+
+
+@pytest.fixture(scope="function", autouse=True)
+def reset_env_vars():
+    # Will reset the environment variables to their original values after each test.
+
+    old_environ = dict(os.environ)
+
+    yield
+
+    os.environ.clear()
+    os.environ.update(old_environ)
```

Comment:
```suggestion
    try:
       yield
    finally:
       os.environ.clear()
       os.environ.update(old_environ)
```
I think is needed so that cleanup is run even on error

Reply:
Good point, thanks!